rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -110,64 +182,135 @@ fn test_get_inputs() {
-            );
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(response, CheckTransactionResponse::Passed));
         }
         WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
     }
+
+    let cycles_after = env.cycle_balance(setup.universal_canister);
+    let expected_cost =
+        CHECK_TRANSACTION_CYCLES_SERVICE_FEE + 2 * get_tx_cycle_cost(INITIAL_BUFFER_SIZE);
+    let actual_cost = cycles_before - cycles_after;
+    // Since we use universal_canister, actual_cost is greater than expected_cost
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < 7_000_000);
+}
+
+#[test]
+fn test_check_transaction_error() {
+    let setup = setup_env();
+    let cycles_before = setup.env.cycle_balance(setup.universal_canister);
+    let mut txid =
+        Txid::from_str("a80763842edc9a697a2114517cf0c138c5403a761ef63cfad1fa6993fa3475ed")
+            .unwrap()
+            .as_ref()
+            .to_vec();
+
+    // Test for cycles not enough
+    let call_id = submit_universal_call(
+        &setup,
+        setup.kyt_canister,
+        "check_transaction",
+        Encode!(&CheckTransactionArgs { txid: txid.clone() }).unwrap(),
+        CHECK_TRANSACTION_CYCLES_REQUIRED - 1,
+    )
+    .expect("submit_call failed to return call id");
+    let result = setup
+        .env
+        .await_call(call_id)
+        .expect("the fetch request didn't finish");
+    match &result {
+        WasmResult::Reply(bytes) => {
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(
+                response,
+                CheckTransactionResponse::NotEnoughCycles
+            ));
+        }
+        WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
+    };
+    let cycles_after = setup.env.cycle_balance(setup.universal_canister);
+    let expected_cost = CHECK_TRANSACTION_CYCLES_SERVICE_FEE;
+    let actual_cost = cycles_before - cycles_after;
+    // Since we use universal_canister, actual_cost is greater than expected_cost
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < 7_000_000);
+
+    // Test for malformatted txid
+    txid.pop();
+    let call_id = submit_universal_call(
+        &setup,
+        setup.kyt_canister,
+        "check_transaction",
+        Encode!(&CheckTransactionArgs { txid }).unwrap(),
+        CHECK_TRANSACTION_CYCLES_REQUIRED,
+    )
+    .expect("submit_call failed to return call id");
+    let result = setup
+        .env
+        .await_call(call_id)
+        .expect("the fetch request didn't finish");
+    match &result {
+        WasmResult::Reply(bytes) => {
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(
+                response,
+                CheckTransactionResponse::Error(msg) if msg.contains("Invalid txid"),
```

Comment:
Can we also check that the cycles minus the service fee are returned?

Reply:
This was checked in the previous test. But sure, I can add it here too.

Reply:
cycle cost test is now added.